sc-memory/sc-memory/src/scs/scs_parser.cpp

```diff
@@ -480,22 +480,24 @@ void Parser::ProcessTriple(
     if (source.IsElementType() && (connector.GetType() == ScType::ConstPermPosArc))
     {
       std::string const & sourceIdtf = source.GetIdtf();
-      ScType const newType = target.m_type | scs::TypeResolver::GetKeynodeType(sourceIdtf);
+      ScType const sourceType = scs::TypeResolver::GetElementType(sourceIdtf);
+      ScType const targetType = target.m_type;
+      ScType const newTargetType = targetType | sourceType;
 
-      if (target.IsElementType() && !newType.CanExtendTo(ScType::ConstNodeClass))
+      if (target.IsElementType() && !newTargetType.CanExtendTo(ScType::ConstNodeClass))
         SC_THROW_EXCEPTION(
             utils::ExceptionParseError,
-            "Can't extend type `" << std::string(target.m_type) << "` to type `" << std::string(newType)
-                                  << "` for element `" << target.GetIdtf() << "`, because element `" << target.GetIdtf()
-                                  << "` is element denoting type of sc-elements.");
-      else if (target.m_type.CanExtendTo(newType))
-        target.m_type = newType;
+            "Can't extend type `" << std::string(targetType) << "` to type `" << std::string(sourceType)
+                                  << "` for specified sc-element, because sc-element `" << target.GetIdtf()
+                                  << "` is sc-element denoting type of sc-elements.");
+      else if (ScType(targetType.BitAnd(~(ScType::Const | ScType::Var))).CanExtendTo(sourceType))
+        target.m_type = newTargetType;
       else
         SC_THROW_EXCEPTION(
             utils::ExceptionParseError,
-            "Can't extend type `" << std::string(target.m_type) << "` to type `" << std::string(newType)
-                                  << "` for element `" << target.GetIdtf() << "`, because `" << std::string(newType) 
-                                  << "` is not subtype of `" << std::string(target.m_type) << "`.");
+            "Can't extend type `" << std::string(targetType) << "` to type `" << std::string(sourceType)
```

Comment:
it'll be more correct to use something like extend type type1 using type type2
because in case of error sourceType is some very general type like MembershipArc and it is not subtype for a lot of types